### PR TITLE
Annotate the type of 'variables' in class All and class None

### PR DESCRIPTION
**Pull request by a-siva**
Annotate the type of 'variables' in class All and class None to ensure that we do not run into https://github.com/dart-lang/sdk/issues/32412

### CODE REVIEW

File: lib/src/all.dart [hunk 1]
```diff
@@ -6,7 +6,7 @@ import '../boolean_selector.dart';
 
 /// A selector that matches all inputs.
 class All implements BooleanSelector {
-  final variables = const [];
+  final List<String> variables = const [];
```
Comment (kevmoo) [inline lib/src/all.dart]: Could this go on the right -> `<String>[]`?

File: pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: boolean_selector
-version: 1.0.3-dev
+version: 1.0.4-dev
```
Comment (kevmoo) [inline pubspec.yaml]: Don't think this is needed, since `1.0.3` has not been released

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 1.0.3
+
+* Annotate the type of 'variables' in class All and class None.
```
Comment (nex3) [inline CHANGELOG.md]: I'd write "Work around a common front-end inference bug." The changelog is intended for downstream users, and as far as they're concerned the only change here is that this now works with the CFE.

File: pubspec.yaml (hunk 2 above)
Comment (nex3) [reply pubspec.yaml]: Yeah, this should just be `1.0.3`.

**Comment by a-siva** [comment]
Thanks

**Pull request merged**